File: src/main/java/org/radarbase/management/service/catalog/SourceTypeResponse.java
package org.radarbase.management.service.catalog;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;

public class SourceTypeResponse {

    @JsonProperty("passive-source-types")
    private List<CatalogSourceType> passiveSources;

    @JsonProperty("active-source-types")
    private List<CatalogSourceType> activeSources;

    @JsonProperty("monitor-source-types")
    private List<CatalogSourceType> monitorSources;

    @JsonProperty("connector-source-types")
    private List<CatalogSourceType> connectorSources;

    public List<CatalogSourceType> getPassiveSources() {
        return passiveSources;
    }

    public List<CatalogSourceType> getActiveSources() {
        return activeSources;
    }

    public List<CatalogSourceType> getMonitorSources() {
        return monitorSources;
    }

    public List<CatalogSourceType> getConnectorSources() {
        return connectorSources;
    }
}

File: src/main/java/org/radarbase/management/service/catalog/SourceTypeResponse.kt
package org.radarbase.management.service.catalog

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true)
class SourceTypeResponse {
    @JsonProperty("passive-source-types")
    val passiveSources: List<CatalogSourceType>? = null

    @JsonProperty("active-source-types")
    val activeSources: List<CatalogSourceType>? = null

    @JsonProperty("monitor-source-types")
    val monitorSources: List<CatalogSourceType>? = null

    @JsonProperty("connector-source-types")
    val connectorSources: List<CatalogSourceType>? = null
}
